cmd/watch.go

```diff
@@ -0,0 +1,214 @@
+			"\n" +
+			"The program to watch is loaded from the project in the current directory by default. Use the `-C` or\n" +
+			"`--cwd` flag to use a different directory.",
+		Args: cmdutil.MaximumNArgs(1),
+		Run: cmdutil.RunResultFunc(func(cmd *cobra.Command, args []string) result.Result {
+			interactive := cmdutil.Interactive()
+			if !interactive {
+				yes = true // auto-approve changes, since we cannot prompt.
+			}
+
+			opts, err := updateFlagsToOptions(interactive, skipPreview, yes)
+			if err != nil {
+				return result.FromError(err)
+			}
+
+			opts.Display = display.Options{
+				Color:                cmdutil.GetGlobalColorization(),
+				ShowConfig:           showConfig,
+				ShowReplacementSteps: showReplacementSteps,
+				ShowSameResources:    showSames,
+				SuppressOutputs:      suppressOutputs,
+				IsInteractive:        interactive,
+				Type:                 display.DisplayWatch,
+				Debug:                debug,
+			}
+
+			return upWorkingDirectory(opts)
+		}),
+	}
+
+	cmd.PersistentFlags().BoolVarP(
+		&debug, "debug", "d", false,
+		"Print detailed debugging output during resource operations")
+	cmd.PersistentFlags().BoolVar(
+		&expectNop, "expect-no-changes", false,
+		"Return an error if any changes occur during this update")
+	cmd.PersistentFlags().StringVarP(
+		&stack, "stack", "s", "",
+		"The name of the stack to operate on. Defaults to the current stack")
+	cmd.PersistentFlags().StringVar(
+		&stackConfigFile, "config-file", "",
+		"Use the configuration values in the specified file rather than detecting the file name")
+	cmd.PersistentFlags().StringArrayVarP(
+		&configArray, "config", "c", []string{},
+		"Config to use during the update")
+	cmd.PersistentFlags().BoolVar(
+		&path, "config-path", false,
+		"Config keys contain a path to a property in a map or list to set")
+	cmd.PersistentFlags().StringVar(
+		&secretsProvider, "secrets-provider", "default", "The type of the provider that should be used to encrypt and "+
+			"decrypt secrets (possible choices: default, passphrase, awskms, azurekeyvault, gcpkms, hashivault). Only"+
+			"used when creating a new stack from an existing template")
+
+	cmd.PersistentFlags().StringVarP(
+		&message, "message", "m", "",
+		"Optional message to associate with the update operation")
+
+	// Flags for engine.UpdateOptions.
+	if hasDebugCommands() {
+		cmd.PersistentFlags().StringSliceVar(
+			&policyPackPaths, "policy-pack", []string{},
+			"Run one or more policy packs as part of this update")
+	}
+	cmd.PersistentFlags().BoolVar(
+		&diffDisplay, "diff", false,
+		"Display operation as a rich diff showing the overall change")
+	cmd.PersistentFlags().IntVarP(
+		&parallel, "parallel", "p", defaultParallel,
+		"Allow P resource operations to run in parallel at once (1 for no parallelism). Defaults to unbounded.")
+	cmd.PersistentFlags().BoolVarP(
+		&refresh, "refresh", "r", false,
+		"Refresh the state of the stack's resources before this update")
```

Comment:
So each update will refresh? I understand wanting to keep these flags similar across `up`, `preview`, and `watch`, but do wonder if this one would be worth cutting here.

Reply:
This particular one actually could definitely be useful for watch - to actively reconcile any changes being made behind the scenes in the cloud provider with what the Pulumi program is doing.  I'll leave this one - but the general point that many of the other flags are't really useful is a good one - I've removed several others.